cli/install/install.go

```diff
@@ -0,0 +1,765 @@
+		}
+	} else {
+		log.Warnf("Downloading tt...")
+		err = downloadRepo("master", logFile, flags.Verbose, path, TTLink)
+	}
+
+	if err != nil {
+		logs, _ := os.ReadFile(logFile.Name())
+		os.Stdout.Write(logs)
+		return err
+	}
+	// Build tt.
+	log.Warnf("Building tt...")
+	err = executeCommand("mage", flags.Verbose, logFile, path,
+		"build")
+	if err != nil {
+		logs, _ := os.ReadFile(logFile.Name())
+		os.Stdout.Write(logs)
+		return err
+	}
+
+	// Copy binary.
+	log.Warnf("Copying executable...")
+	if _, err := os.Stat(dst); os.IsNotExist(err) {
+		err = os.Mkdir(dst, 0777)
+		if err != nil {
+			return fmt.Errorf("Unable to create bindir: %s", err)
+		}
+	}
+	if flags.Reinstall {
+		os.Remove(dst + "/" + version)
+	}
+	err = util.CopyFilePreverse(path+"/tt", dst+"/"+version)
+	if err != nil {
+		return err
+	}
+
+	// Set symlink.
+	err = changeSymlink(version, dst, "tt")
+	log.Warnf("Done.")
+	if flags.Noclean {
+		log.Warnf("Artifacts can be found at: " + path)
+	}
+	return err
+}
+
+// installTarantool installs selected version of tarantool.
+func installTarantool(version string, bindst string, headerDir string, flags InstallationFlag,
+	local string) error {
+	tarVersion := ""
+
+	// Get latest tarantool tag if needed.
+	if version == "tarantool" {
+		tarVersion, _ = getLatestTarantoolTag()
+		version += "=" + tarVersion
+	} else {
+		tarVersion = strings.TrimPrefix(version, "tarantool=")
+	}
+	if util.IsDeprecated(tarVersion) {
+		return fmt.Errorf("Deprecated version of tarantool")
+	}
+
+	// Check bin and header dirs.
+	if bindst == "" {
+		return fmt.Errorf("Cannot procces bindir.")
+	}
+	if headerDir == "" {
+		return fmt.Errorf("Cannot procces headerdir.")
+	}
+	logFile, _ := ioutil.TempFile("", "tarantool_install")
+
+	log.Warn("Installing tarantool=" + tarVersion)
+
+	// Check if programm is already installed.
+	log.Warnf("Checking existing...")
+	if checkExisting(version, bindst) && !flags.Reinstall {
+		log.Warnf("That version binary already exists, changing symlinc...")
+		err := changeSymlink(version, bindst, "tarantool")
+		log.Warnf("Done")
+		return err
+	}
+
+	if checkExisting(version, headerDir) && !flags.Reinstall {
+		log.Warnf("That version headers already exists, changing symlinc...")
+		err := changeSymlink(version, headerDir, "tarantool")
+		log.Warnf("Done")
+		return err
+	}
+
+	// Check dependencies.
+	log.Warnf("Checking dependecies...")
+	if flags.Force {
+		log.Warnf("Skipped...")
+	} else if !checkTarantoolDependencies() {
+		return nil
+	}
+
+	path, err := os.MkdirTemp("", "tarantool_install")
+	os.Chmod(path, 0777)
+	if err != nil {
+		return err
+	}
+	if !flags.Noclean {
+		defer os.RemoveAll(path)
+	}
+
+	// Download tarantool.
+	if flags.Local {
+		log.Warnf("Checking local files...")
+		if checkExisting("tarantool", local) {
+			log.Warnf("Local files found, installing from them...")
+			localPath, _ := util.JoinAbspath(local, "tarantool")
+			err = util.CopyDir(localPath, path)
```

Comment:
There is already a copy function in `github.com/otiai10/copy`, which is our dependency. Is there any point to have an own function for this purpose?

Reply:
We had a conversation with @LeonidVas that if we can write something with our hands, we don't need to use additional dependencies. Also, it is dependencie only of catridge-cli and some tests, so I didn't really see a purpose to change it.

Reply:
Seems to be fixed.